src/modules/graphql/resolvers/class.resolver.ts

```diff
@@ -35,17 +63,34 @@ export class ClassResolver {
     }
 
     @Mutation('createClass')
-    async createClass(obj, { class: newClass }) {
+    async createClass(obj, { class: newClass }, context) {
+        checkAndGetBasePermission(Get.getObject(context, 'user'), DBOperation.CREATE, Asset.CLASS);
         return this.classPersistence.createClass(newClass);
     }
 
     @Mutation('updateClass')
-    async updateClass(obj, {id, class: classObj}) {
+    async updateClass(obj, {id, class: classObj}, context) {
+        const permission = checkAndGetBasePermission(Get.getObject(context, 'user'), DBOperation.UPDATE, Asset.CLASS);
+        if (permission === Permission.OWN) {
+            const user: UserDbModel = await this.userPersistenceService.getById(context.user.id);
+            const userClassId = user.class_id ? user.class_id.toString() : '';
+            if (userClassId !== id.toString()) {
+                throw new Error(NO_PERMISSION);
+            }
+        }
         return this.classPersistence.updateClass(id, classObj);
     }
 
     @Mutation('deleteClass')
-    async deleteClass(obj, {id}) {
+    async deleteClass(obj, {id}, context) {
+        const permission = checkAndGetBasePermission(Get.getObject(context, 'user'), DBOperation.DELETE, Asset.CLASS);
+        if (permission === Permission.OWN) {
+            const user: UserDbModel = await this.userPersistenceService.getById(context.user.id);
+            const userClassId = user.class_id ? user.class_id.toString() : '';
+            if (userClassId !== id) {
```

Comment:
In updateClass we convert the id to string and in the deleteClass we are not. Is it in purpose?

Reply:
fixed